### ISSUE TITLE
strange crash on linux in latest (v2.1) with steps to reproduce

### ISSUE THREAD

**Issue by CapsAdmin** [LuaJIT/LuaJIT#131]
In vulkan_test.lua luajit will crash when glfw.Init() is called.

Okay so something might be wrong with `glfw.Init()` but it turns out if I remove `local vk = require("libvulkan")` it won't crash and `glfw.Init()` runs fine.

**So I add `do return end` at the top of libvulkan.lua but it still crashes. However if I comment out everything it won't crash.**

```
0  0x00007ffff70d3366 in ?? ()
1  0x0000000000425189 in lj_vm_ffi_call ()
2  0x0000000000452390 in lj_ccall_func (L=L@entry=0x40000378, cd=<optimized out>) at lj_ccall.c:985
3  0x0000000000420b96 in lj_cf_ffi_meta___call (L=0x40000378) at lib_ffi.c:230
4  0x00000000004231eb in lj_BC_FUNCC ()
5  0x000000000040f289 in lua_pcall (L=L@entry
[luajitcrash.zip](https://github.com/LuaJIT/LuaJIT/files/136134/luajitcrash.zip)
=0x40000378, nargs=nargs@entry=0, nresults=-1, errfunc=errfunc@entry=2) at lj_api.c:1055
6  0x00000000004043fc in docall (L=0x40000378, narg=0, clear=0) at luajit.c:121
7  0x000000000040524e in handle_script (n=<optimized out>, argv=<optimized out>, L=<optimized out>) at luajit.c:288
8  pmain (L=0x40000378) at luajit.c:537
9  0x00000000004231eb in lj_BC_FUNCC ()
10 0x000000000040f343 in lua_cpcall (L=L@entry=0x40000378, func=func@entry=0x4049e0 <pmain>, ud=ud@entry=0x0) at lj_api.c:1079
11 0x0000000000403f14 in main (argc=2, argv=0x7fffffffde18) at luajit.c:565
```

I've compiled both glfw and luajit (latest from git) with debug symbols. ffi.load("vulkan") loads libvulkan.so from the LunarG SDK. I didn't add it because it's not even executed in this test case.

run `bash run.bash` to test

**Comment by CapsAdmin** [comment]
forgot to add the test
[luajitcrash.zip](https://github.com/LuaJIT/LuaJIT/files/136156/luajitcrash.zip)

**Comment by luastoned** [comment]
I modified CapsAdmin's test on Windows and got the crash too, see crash.lua for details.

    local glfw = require("libglfw")
    require("libdummy") -- alllocate a dummy table, basically what vulkan did
    glfw.glfwInit()

If you don't require GLFW but define it in the crash file it will not crash.

    ffi.cdef("int glfwInit();")
    local glfw = ffi.load("glfw3")
    require("libdummy")
    glfw.glfwInit()

I stripped GLFW of all functions except Init, which seems to crash if called after the dummy require.

[luajitcrash.zip](https://github.com/LuaJIT/LuaJIT/files/137955/luajitcrash.zip)
